src/urllib3/util/request.py

```diff
@@ -123,7 +132,7 @@ def set_file_position(
     return pos
 
 
-def rewind_body(body: IO[AnyStr], body_pos: Optional[Union[int, object]]) -> None:
+def rewind_body(body: IO[AnyStr], body_pos: _TYPE_FAILEDTELL) -> None:
```

Comment:
I think we may still want the `Union` here and replace the generic `object` with the type of the new sentinel type.